hawc/tools/tables/sot.py

```diff
@@ -0,0 +1,174 @@
+from enum import Enum
+from typing import List, Optional
+
+import pandas as pd
+from pydantic import BaseModel, Field, conint
+
+from hawc.apps.animal.models import Endpoint
+from hawc.apps.materialized.models import FinalRiskOfBiasScore
+from hawc.apps.riskofbias.constants import SCORE_SHADES, SCORE_SYMBOLS
+
+from .base import BaseCell, BaseCellGroup, BaseTable
+from .generic import GenericCell
+from .parser import QuillParser, color_background, tag_wrapper
+
+
+class AttributeChoices(Enum):
+    RobScore = "rob_score"
+    StudyName = "study_name"
+    SpeciesStrainSex = "species_strain_sex"
+    Doses = "doses"
+    FreeHtml = "free_html"
+
+    def get_rob_score(self, selection: pd.DataFrame) -> BaseCell:
+        values = selection["score_score"].values
+
+        judgment = 0 if values.size == 0 else values[0]
+        return JudgmentCell(judgment=judgment)
+
+    def get_study_name(self, selection: pd.DataFrame) -> BaseCell:
+        values = selection["study citation"].dropna().unique()
+
+        text = "; ".join(values)
+        return GenericCell(quill_text=tag_wrapper(text, "p"))
+
+    def get_species_strain_sex(self, selection: pd.DataFrame) -> BaseCell:
+        values = selection["animal description"].dropna().unique()
+
+        text = "; ".join(values)
+        return GenericCell(quill_text=tag_wrapper(text, "p"))
+
+    def get_doses(self, selection: pd.DataFrame) -> BaseCell:
+        values = selection["doses"].dropna().unique()
+
+        text = "; ".join(values)
+        return GenericCell(quill_text=tag_wrapper(text, "p"))
+
+    def get_free_html(self, selection: pd.DataFrame) -> BaseCell:
+        # this will be overridden by 'customized' html
+        return GenericCell()
+
+    def get_cell(self, selection: pd.DataFrame) -> BaseCell:
+        return getattr(self, f"get_{self.value}")(selection)
+
+
+class Subheader(BaseModel):
+    label: str
+    start: conint(ge=0)
+    length: conint(ge=1)
+
+
+class Column(BaseModel):
```

Comment:
we probably need to add a width here, with a default value of 1 for all to be equal widths?

Reply:
to close the loop on our chat, feel free to use whatever existing mechanisms we already have to solve this. it just seemed like a good home on the current data structures